node/src/components/block_synchronizer/tests.rs

```diff
@@ -95,35 +180,77 @@ fn check_sync_global_state_event(event: MockReactorEvent, block: &Block) {
     );
 }
 
+// Calls need_next for the block_synchronizer and processes the effects resulted returning a list of
+// the new events that were generated
+async fn need_next(
+    rng: &mut TestRng,
+    reactor: &MockReactor,
+    block_synchronizer: &mut BlockSynchronizer,
+    num_expected_events: usize,
+) -> Vec<MockReactorEvent> {
+    let effects = block_synchronizer.need_next(reactor.effect_builder(), rng);
+    assert_eq!(effects.len(), num_expected_events);
+    reactor.process_effects(effects).await
+}
+
+fn register_multiple_signatures<'a, I: IntoIterator<Item = &'a Arc<SecretKey>>>(
+    builder: &mut BlockBuilder,
+    block: &Block,
+    validator_keys_iter: I,
+) {
+    for secret_key in validator_keys_iter {
+        // Register a finality signature
+        let signature = FinalitySignature::create(
+            *block.hash(),
+            block.header().era_id(),
+            secret_key.as_ref(),
+            PublicKey::from(secret_key.as_ref()),
+        );
+        assert!(signature.is_verified().is_ok());
+        assert!(builder.register_finality_signature(signature, None).is_ok());
+    }
+}
+
+impl BlockSynchronizer {
+    // Create an initialized block synchronizer with default config and maximum 5 simultaneous peers
```

Comment:
A nit (don't hardcode magic number) :)

```suggestion
    // Create an initialized block synchronizer with default config and maximum simultaneous peers
```